frappe/core/doctype/user/user.js

```diff
@@ -76,6 +76,28 @@ frappe.ui.form.on('User', {
 				}
 			}
 		}
+		if (frm.doc.user_emails){
+			var found =0
+			for (var i = 0;i<frm.doc.user_emails.length;i++){
+				if (frm.doc.email==frm.doc.user_emails[i].email_id){
+					found = 1;
+				}
+			}
+			frm.get_field("create_user_email").df.hidden = found;
+			frm.refresh_field("create_user_email");
+		}else{
+			frm.get_field("create_user_email").df.hidden = 1
+		}
+		frm.refresh_field("create_user_email");
+
+		if (frappe.route_titles["unsaved"]===1){
```

Comment:
this is for the email matching so u can create a contact fill in customer or supplier and it will redirect back to page